File: BeveragePaymentApi/BeveragePaymentApi/Dto/Beverage.cs


namespace Dto;


public class BeverageDto
{
    public string? BeverageId {get; set;}

    public string? Name { get; set; }

    public string? Description { get; set; }

    public string? ImageSrc { get; set; }

    public double BasePrice { get; set; }

    public double MaxPrice { get; set; }

    public double MinPrice { get; set; }
}

File: frontend/packages/api/src/types/beverage.ts
/** @format */

interface Price {
  id: number;
  amount: number;
  timestamp: Date;
  beverageId: number;
}

interface Beverage {
  beverageId: number;
  name: string;
  description: string;
  imageSrc: string;
  basePrice: number;
  minPrice: number;
  maxPrice: number;
  isActive: boolean;
  buyMultiplier: number;
  halfTime: number;
  totalSales: number;
  prices: Price[];
}

interface BeverageDTO {
  name: string;
  description: string;
  //imageSrc: string;
  basePrice: number;
  minPrice: number;
  maxPrice: number;
}

export type { Beverage, Price, BeverageDTO };
